src/Symfony/Bundle/TwigBundle/CacheWarmer/TemplateCacheWarmer.php

```diff
@@ -26,21 +26,26 @@
  */
 class TemplateCacheWarmer implements CacheWarmerInterface, ServiceSubscriberInterface
 {
-    private ContainerInterface $container;
     private Environment $twig;
-    private iterable $iterator;
 
-    public function __construct(ContainerInterface $container, iterable $iterator)
+    public function __construct(private ContainerInterface $container, private iterable $iterator, private string $cacheFolder = 'twig')
     {
-        // As this cache warmer is optional, dependencies should be lazy-loaded, that's why a container should be injected.
-        $this->container = $container;
-        $this->iterator = $iterator;
+        if (\func_num_args() < 3) {
+            trigger_deprecation('symfony/twig-bundle', '7.1', 'The "string $cacheFolder" argument of "%s()" method will not be optional anymore in version 8.0, not defining it is deprecated.', __METHOD__);
```

Comment:
we don't have to make it required, we can set it to 'twig' by default

Reply:
Deprecation removed